data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/IdsMultipartSender.java

```diff
@@ -312,4 +323,20 @@ private IdsMultipartParts extractResponseParts(ResponseBody body) throws Excepti
                 .build();
     }
 
+    private R checkResponseType(R response) {
+        var type = getAllowedResponseTypes();
+
+        Objects.requireNonNull(type);
+
+        if (response instanceof MultipartResponse) {
+            var header = ((MultipartResponse<?>) response).getHeader();
```

Comment:
I was wondering why the `R` type does not extends `MultipartResponse`, so I have a question, why `Catalog` is not enclosed in a `MultipartResponse` object? This would make the signature of the `IdsMultipartSender` more readable and this cast could be avoided, or am I missing something?

Reply:
Currently , the method isn't returning a `MultipartResponse` because the `CatalogServiceImpl` and `BatchedRequestFetcher` expect a `Catalog` as return type for further processing, not just an `Object`.

Reply:
@juliapampus couldn't be the `IdsRemoteMessageDispatcher` that takes care of extracting the `Catalog` or every other object wrapped into `MultipartResponse` and returning it?

Reply:
hmmm.. this could help, but how should the services then be able to implement error handling? Currently, the future is evaluated in almost every service and depending on throwables, responses etc. process continue differently. Even with using some kind of result object, the services expecting a specific object would have to be changed.

Reply:
okay, so as we have this `AbstractResult` class. What about creating a new `MessageResult` and adapting the services that use this message dispatcher results? @ndr-brt

Reply:
The dispatcher could then also, as proposed, take care of extracting the wrapped object and returning it in `success()`.

Reply:
@juliapampus why don't handle directly everything with the `CompletableFuture`? It's a "wrapping" type as `Result`.
The services want to receive a `CompletableFuture` with a certain content (e.g. `Catalog`), while the `MultipartResponse` it's a knowledge that should stay in the "multipart" world (if another protocol would exists, probably there would be another class wrapping the response).
So in the `IdsRemoteMessageDispatcher` the final line could be:
```java
return handler.send(message, context).thenApply(it -> (T) it.getPayload())
```
the `thenApply` works as a `map` function, it modifies the content of the wrapping type, in this case asynchronously

Reply:
@ndr-brt Agree, but we should also not move multipart logic to the generic `IdsRemoteMessageDispatcher`. I therefore added the unpacking of the `MultipartResponse` to the `IdsMultipartSender` and changed the signature of the sender classes to return a simple object, not wrapped in a `MultipartResponse`.

Reply:
I am not sure what the benefits may be, but as we currently want to make things easier, I would not introduce new interfaces. So I would keep this PR as-is and move to the next issue (#1697) as soon as this here is merged.

Reply:
@juliapampus fine for me